tests/providers/hashicorp/secrets/test_vault.py

```diff
@@ -0,0 +1,118 @@
+        mock_client = mock.MagicMock()
+        mock_hvac.Client.return_value = mock_client
+        mock_client.secrets.kv.v2.read_secret_version.return_value = {
+            'request_id': '94011e25-f8dc-ec29-221b-1f9c1d9ad2ae',
+            'lease_id': '',
+            'renewable': False,
+            'lease_duration': 0,
+            'data': {
+                'data': {'AIRFLOW_CONN_TEST_POSTGRES': 'postgresql://airflow:airflow@host:5432/airflow'},
+                'metadata': {'created_time': '2020-03-16T21:01:43.331126Z',
+                             'deletion_time': '',
+                             'destroyed': False,
+                             'version': 1}},
+            'wrap_info': None,
+            'warnings': None,
+            'auth': None
+        }
+
+        kwargs = {
+            "path": "test_path",
+            "auth_type": "token",
+            "url": "http://127.0.0.1:8200",
+            "token": "s.7AU0I51yv1Q1lxOIg1F3ZRAS"
+        }
+
+        test_client = VaultSecrets(**kwargs)
+        returned_uri = test_client.get_conn_uri(conn_id="test_postgres")
+        self.assertEqual('postgresql://airflow:airflow@host:5432/airflow', returned_uri)
+
+    @mock.patch.dict('os.environ', {
+        'AIRFLOW_CONN_TEST_MYSQL': 'mysql://airflow:airflow@host:5432/airflow',
+    })
+    @mock.patch("airflow.providers.hashicorp.secrets.vault.hvac")
+    def test_get_conn_uri_non_existent_key(self, mock_hvac):
+        """
+        Test that if the key with connection ID is not present in Vault, VaultClient.get_connections
+        should return None and fallback to the environment variable if it is set
+        """
+        mock_client = mock.MagicMock()
+        mock_hvac.Client.return_value = mock_client
+        # Response does not contain the requested key
+        mock_client.secrets.kv.v2.read_secret_version.return_value = {
+            'request_id': '94011e25-f8dc-ec29-221b-1f9c1d9ad2ae',
+            'lease_id': '',
+            'renewable': False,
+            'lease_duration': 0,
+            'data': {
+                'data': {'AIRFLOW_CONN_TEST_POSTGRES': 'postgresql://airflow:airflow@host:5432/airflow'},
+                'metadata': {'created_time': '2020-03-16T21:01:43.331126Z',
+                             'deletion_time': '',
+                             'destroyed': False,
+                             'version': 1}},
+            'wrap_info': None,
+            'warnings': None,
+            'auth': None
+        }
+
+        kwargs = {
+            "path": "test_path",
+            "auth_type": "token",
+            "url": "http://127.0.0.1:8200",
+            "token": "s.7AU0I51yv1Q1lxOIg1F3ZRAS"
+        }
+
+        test_client = VaultSecrets(**kwargs)
+        self.assertIsNone(test_client.get_conn_uri(conn_id="test_mysql"))
+        self.assertIsNone(test_client.get_connections(conn_id="test_mysql"))
+
+        # Fallback to connection defined in Environment Variable
```

Comment:
Isn't this test (re)testing the base secrets interface?

Reply:
Fixed

Reply:
The test didn't previously exist to check this behavior in the base too. This is now added and merged, so will remove